### PR TITLE
ae-address redesign

### PR DESCRIPTION
**Pull request by SamLife91**
- change design of aeAdddress component
- add addition unit tests

### CODE REVIEW

**Comment by acdbaykal** [comment]
Should the version below also be supported?

[](https://postimg.org/image/5g4266lqf/)

**Comment by acdbaykal** [comment]
Same question for the next one

[](https://postimg.org/image/lfmppqphz/)

**Comment by acdbaykal** [comment]
Than there is also one with dots at the instead of the middle

[](https://postimg.org/image/wghuuohpz/)

File: src/components/aeAddress/aeAddress.vue [hunk 1]
```diff
@@ -1,9 +1,26 @@
 <template>
-  <div :class='cssClass'>
-    <ae-identity-avatar v-if='showAvatar' :address='address' />
-    <span class='address'>{{displayAddress}}</span>
+  <div :class='cssClass' class="aeAddressWrapper">
```
Comment (acdbaykal) [inline src/components/aeAddress/aeAddress.vue]: I'm really not a fan of using :class and class on the same element. I prefer something like

```
:class="['aeAddressWrapper', cssClass]"
```

File: src/components/aeAddress/aeAddress.vue [hunk 2]
```diff
@@ -1,9 +1,26 @@
...
+          {{chunk}}
+        </div>
+      </div>
+    </div>  
   </div>
 </template>
 <script src='./aeAddress.js'/>
 /* eslint no-unused-expressions: "off" */
-<style src='./aeAddress.scss'/>
+<style scoped="scoped" src='./aeAddress.scss'/>
```
Comment (acdbaykal) [inline src/components/aeAddress/aeAddress.vue]: Good you cached that!

```
<style scoped src='./aeAddress.scss'/>
```
is sufficient though

**Comment by dadaphl** [comment]
@raymundjacobs 
Is it true, that we need a AeAddress with the ellipsis `...` in the end? Maybe we can avoid this and rethink this specific screen.

**Comment by dadaphl** [comment]
I suggest that the AeAddress component stays simple by default. The new layout should be a variation.

File: src/components/aeAddress/aeAddress.vue (hunk 1 above)
Comment (acdbaykal) [reply src/components/aeAddress/aeAddress.vue]: No, not that I know. Maybe Vue is slightly more performant one variation Idk. My reason is that I like to have it in one place.

**Comment by SamLife91** [comment]
@acdbaykal @dadaphl @raymundjacobs  i can implement this as additional (optional) layout or type

**Pull request merged**